features/largeVideo/components/LargeVideo.js

```diff
@@ -63,9 +64,14 @@ class LargeVideo extends Component {
             this.props.dispatch(participantSelected(activeParticipant.id));
         }
 
+        let mirrored = videoTrack &&
+            videoTrack.isLocal() &&
+            videoTrack.videoType === 'camera';
```

Comment:
That's duplicated logic. Usually, it's a signal to me that there may be a need to introduce an abstraction. The simplest 1 is a function. There may be an opportunity for a slightly more rewarding abstraction though such as ParticipantVideo or TrackVideo. I believe we should proactively look back at our existing source code in the light of such signals & refactor for reusability, consistency, etc.

For example, this duplication provokes me to look at both VideoThumbnail & LargeVideo & ask: why have to different approaches to mirrored? 1 is remembering it in state & the other is recomputing on render. I find such differences to burden the readability & maintainability of the source code.

Another issue with this duplication is the use of 'camera' i.e. a literal. If duplication was eliminated, the effect of such a bad programming practice would have been halved.